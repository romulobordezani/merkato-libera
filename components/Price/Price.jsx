import React from 'react';
import PropTypes from 'prop-types';

import styles from './Price.scss';

const Price = ({ fontSize, item }) => {
  const { price } = item;
  return (
    <span className={styles.price} style={{ fontSize }}>
      <span className={styles.price_currency}>{price.currency}</span>
      <span className={styles.price_amount}>{price.amount.toLocaleString('de-DE')}</span>
      <span className={styles.price_decimals}>{price.decimals}</span>
    </span>
  );
};

Price.propTypes = {
  fontSize: PropTypes.string,
  item: PropTypes.object.isRequired
};

Price.defaultProps = {
  fontSize: '36px'
};

export default Price;
