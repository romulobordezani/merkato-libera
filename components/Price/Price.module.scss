@import "../_SassLib/breakpoints";

.price {
  font-size: 46px;
  padding: 32px 0;
  font-weight: 200;
  display: inline-block;

  @include breakpoint($small) {
    padding: 6px 0;
  }

  &_currency {
    margin-right: 12px;
  }

  &_amount {

  }

  &_decimals {
    font-weight: 100;
    font-size: 0.5em;
    vertical-align: super;
    margin-left: 3px;
  }
}
