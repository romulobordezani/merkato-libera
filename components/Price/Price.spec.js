import React from 'react';
import renderer from 'react-test-renderer';

import Price from './Price';

const ItemMock = require('../../__mocks__/Item');

describe('Price', () => {
  it('Snapshot renders default', () => {
    expect(renderer.create(<Price item={ItemMock} />)).toMatchSnapshot();
  });
});
