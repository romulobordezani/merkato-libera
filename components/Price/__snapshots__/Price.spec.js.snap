// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Snapshot renders default 1`] = `
<span
  style={
    Object {
      "fontSize": "36px",
    }
  }
>
  <span>
    $
  </span>
  <span>
    1.800
  </span>
  <span>
    
  </span>
</span>
`;
